test-compile/compiles/literals_test.cpp

```diff
@@ -0,0 +1,9 @@
+#include <boost/safe_float/literals.hpp>
```

Comment:
It will be nice to also test that wrong numbers fails. You can have an example in this project to see how cmake defines test where we test that something compiles and test that we test it fails to compile because of an static-assert or similar: 
https://github.com/sdavtaker/cadmium/tree/master/test-compile